File: CockyScript.js
var request = require('request'),
    cheerio = require('cheerio'),
    YTsuscriptors, YTviews,
    TVfollowers, TVdesigns, TVdesignspages, TVurls = [], TVlikes = [], TVcomments = [], TVcollects = [], TVmakes = [], TVviews = [], TVdownloads = [], TVcounter = 1, TVlikesCounter = 0, TVcommentsCounter = 0, TVcollectsCounter = 0, TVmakesCounter = 0, TVviewsCounter = 0, TVdownloadsCounter = 0,
    TWfollowers,
    HDfollowers, HDlikes,
    GHurls=[], GHcounter=0, GHcontributors=[], GHwatchers=[], GHstars=[], GHforks=[], GHcontributorsCounter=0, GHwatchersCounter=0, GHstarsCounter=0, GHforksCounter=0,
    GGmembers,
    IGfollowers;
var youtubeUrl="https://www.youtube.com/user/ALM37454/about",
    twitterUrl="https://twitter.com/_AngelLM",
    hackadayUrl="https://hackaday.io/AngelLM",
    thingiverseUrl="https://www.thingiverse.com/AngelLM/designs/",
    githubUrl="https://github.com/AngelLM?utf8=%E2%9C%93&tab=repositories&q=&type=source&language=",
    googlegroupsUrl="https://groups.google.com/forum/#!aboutgroup/thor-opensource-3d-printable-robotic-arm",
    instagramUrl="https://www.instagram.com/angel_lm_/";




request(youtubeUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        YTsuscriptors = $('.about-stat','#browse-items-primary').eq(0).text().replace(/[^0-9.]/g, '');
        YTviews = $('.about-stat','#browse-items-primary').eq(1).text().replace(/[^0-9]/g, '');
        console.log("-------------");
        console.log("YouTube stats");
        console.log("-------------");
        console.log("Suscriptors: " + YTsuscriptors);
        console.log("Views: " + YTviews);
        console.log("");
    }
});

request(twitterUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        TWfollowers = $('.ProfileNav-stat','.ProfileNav-item--followers').text().replace(/[^0-9.]/g, '');
        console.log("-------------");
        console.log("Twitter stats");
        console.log("-------------");
        console.log("Followers: " + TWfollowers);
        console.log("");
    }
});


request(hackadayUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        HDfollowers = $('.followers').text().replace(/[^0-9.]/g, '');
        HDlikes = $('.follows').text().replace(/[^0-9.]/g, '');
        console.log("-------------");
        console.log("Hackaday stats");
        console.log("-------------");
        console.log("Followers: " + HDfollowers);
        console.log("Likes: " + HDlikes);
        console.log("");
    }
});



first_TV();
first_GH();

request(googlegroupsUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        var GGmembersID = getAllElementsWithText($(this)).attributes.for.value;
        //getElementById(getAllElementsWithText().attributes.for.value).textContent;
        console.log("-------------");
        console.log("Google Groups stats");
        console.log("-------------");
        console.log("Members: " + GGmembers);
        console.log("");
    }
});

/*request(instagramUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        IGfollowers = $("._bkw5z").innerHTML;
        console.log("-------------");
        console.log("Instagram stats");
        console.log("-------------");
        console.log("Followers: " + IGfollowers);
        console.log("");
    }
});*/

//Thingiverse Functions

function first_TV(){
  request(thingiverseUrl, function(err, resp, body){
    if(!err && resp.statusCode == 200){
        var $ = cheerio.load(body);
        TVdesigns = $('.user-count','#main').eq(2).text().replace(/[^0-9.]/g, '');
        TVdesignspages=Math.ceil(TVdesigns/12);
        TVfollowers = $('.user-count','#main').eq(0).text().replace(/[^0-9.]/g, '');
        second_TV();
      }
    });
}

function second_TV(){
          var newurl = thingiverseUrl.concat("page:" + TVcounter);
          request(newurl, function(err, resp, body){
            if(!err && resp.statusCode == 200){
                var $ = cheerio.load(body);
                $('.thing-img-wrapper','#profile-content').each(function(){
                    var thisurl = $(this).attr('href').toString();
                    TVurls.push(thisurl);
                });
                TVcounter+= 1;
              }
              if (TVcounter>TVdesignspages){
                  TVcounter=0;
                  third_TV();
              }
              else {
                  second_TV();
              }
            });

        }

function third_TV(){
          var preurl = "https://www.thingiverse.com";
          var fullurl = preurl.concat(TVurls[TVcounter]);
          request(fullurl, function(err, resp, body){
              if(!err && resp.statusCode == 200){
                  var $ = cheerio.load(body);
                  var TVview = Number($('.thing-views','#thing-description').text().replace(/[^0-9.]/g, ''));
                  TVviews.push(TVview);
                  var TVdownload = Number($('.thing-downloads','#thing-description').text().replace(/[^0-9.]/g, ''));
                  TVdownloads.push(TVdownload);
                  var TVlike = Number($('.thing-like','#thing-page').eq(0).text().replace(/[^0-9.]/g, ''));
                  TVlikes.push(TVlike);
                  var TVcollect = Number($('.thing-collect','#thing-page').eq(0).text().replace(/[^0-9.]/g, ''));
                  TVcollects.push(TVcollect);
                  var TVcomment = Number($('.thing-comment','#thing-page').eq(0).text().replace(/[^0-9.]/g, ''));
                  TVcomments.push(TVcomment);
                  var TVmake = Number($('.thing-made','#thing-page').eq(1).text().replace(/[^0-9.]/g, ''));
                  TVmakes.push(TVmake);
                  TVcounter+= 1;
              }
              if (TVcounter==TVurls.length){
                  TVcounter= 0;
                  forth_TV();
              }
              else {
                  third_TV();
              }
        });
  }

  function forth_TV(){
    TVlikesCounter += TVlikes[TVcounter];
    TVcommentsCounter += TVcomments[TVcounter];
    TVcollectsCounter += TVcollects[TVcounter];
    TVmakesCounter += TVmakes[TVcounter];
    TVviewsCounter += TVviews[TVcounter];
    TVdownloadsCounter += TVdownloads[TVcounter];
    TVcounter+= 1;
    if(TVcounter==TVlikes.length){
        console.log("-------------");
        console.log("Thingiverse stats");
        console.log("-------------");
        console.log("Followers: " + TVfollowers);
        console.log("Views: " + TVviewsCounter);
        console.log("Downloads: " + TVdownloadsCounter);
        console.log("Likes: " + TVlikesCounter);
        console.log("Collects: " + TVcollectsCounter);
        console.log("Makes: " + TVmakesCounter);
        console.log("");
      }
    else {
      forth_TV();
    }
  }

//GitHub Functions

function first_GH(){
    request(githubUrl, function(err, resp, body){
        if(!err && resp.statusCode == 200){
            var $ = cheerio.load(body);
            $('.col-12').each(function(){
                var thisGHurl = $('a',this).attr('href');
                GHurls.push(thisGHurl);
            });
          second_GH();
        }
    });
}

function second_GH(){
    var GHpreurl = "https://github.com";
    var GHfullurl = GHpreurl.concat(GHurls[GHcounter]);
    request(GHfullurl, function(err, resp, body){
        if(!err && resp.statusCode == 200){
            var $ = cheerio.load(body);
            var GHwatch = Number($('.social-count','.repohead-details-container').eq(0).text().replace(/[^0-9.]/g, ''));
            GHwatchers.push(GHwatch);
            var GHstar = Number($('.social-count','.repohead-details-container').eq(1).text().replace(/[^0-9.]/g, ''));
            GHstars.push(GHstar);
            var GHfork = Number($('.social-count','.repohead-details-container').eq(2).text().replace(/[^0-9.]/g, ''));
            GHforks.push(GHfork);
            var GHcontributor = Number($('.num','.numbers-summary').eq(3).text().replace(/[^0-9.]/g, ''));
            GHcontributors.push(GHcontributor);
            GHcounter+= 1;
        }
        if (GHcounter==GHurls.length){
            GHcounter= 0;
            third_GH();
        }
        else {
            second_GH();
        }
    });

}

function third_GH(){
    GHcontributorsCounter += GHcontributors[GHcounter];
    GHwatchersCounter += GHwatchers[GHcounter];
    GHstarsCounter += GHstars[GHcounter];
    GHforksCounter += GHforks[GHcounter];
    GHcounter+= 1;
    if(GHcounter==GHcontributors.length){
        console.log("-------------");
        console.log("GitHub stats");
        console.log("-------------");
        console.log("Watchers: " + GHwatchersCounter);
        console.log("Stars: " + GHstarsCounter);
        console.log("Forks: " + GHforksCounter);
        console.log("contributors: " + GHcontributorsCounter);
        console.log("");
    }
    else {
        third_GH();
    }
}

function getAllElementsWithText(thiso)
{
  var matchingElements = [];
  var allElements = $(thiso).getElementsByTagName('*');
  for (var i = 0, n = allElements.length; i < n; i++)
  {
    if (allElements[i].textContent == " Miembros ")
    {
      return allElements[i];
    }
  }
}

File: CockyScript.py
import sys
sys.path.append("functions")
import thingiverse
import github
import hackaday
import youtube
import twitter
import googlegroups
import instagram

# Users & Names variables. Fill them with your usernames if you want the stats of that platform. Leave it blank if you don't want the stats.

TVuser = 'AngelLM'                                      # Thingiverse username. https://www.thingiverse.com/USERNAME/
GHuser = 'AngelLM'                                      # GitHub username. https://github.com/USERNAME
HDuser = 'AngelLM'                                      # Hackaday username. https://hackaday.io/USERNAME
YTuser = 'AngelLM'                                      # Youtube username. https://www.youtube.com/USERNAME
TWuser = '_AngelLM'                                     # Twitter username. https://twitter.com/USERNAME
GGname = 'thor-opensource-3d-printable-robotic-arm'     # Google Group name. https://groups.google.com/forum/#!forum/GROUPNAME
IGuser = 'angel_lm_'                                    # Instagram username. https://www.instagram.com/

if TVuser != '':
    TVstats = thingiverse.getStats(TVuser)
    print '----------------------'
    print 'Thingiverse Stats'
    print '----------------------'
    print 'Total views: ' + str(TVstats[0])
    print 'Total downloads: ' + str(TVstats[1])
    print 'Total likes: ' + str(TVstats[2])
    print 'Total collects: ' + str(TVstats[3])
    print 'Total makes: ' + str(TVstats[4])
    print 'Total remixes: ' + str(TVstats[5])
    print 'Total comments: ' + str(TVstats[6])

if GHuser != '':
    GHstats = github.getStats(GHuser)
    print ''
    print '----------------------'
    print 'GitHub Stats'
    print '----------------------'
    print 'Total followers: ' + str(GHstats[0])
    print 'Total watchers: ' + str(GHstats[1])
    print 'Total stars: ' + str(GHstats[2])
    print 'Total forks: ' + str(GHstats[3])
    print 'Total contributors: ' +str(GHstats[4])

if HDuser != '':
    HDstats = hackaday.getStats(HDuser)
    print ''
    print '----------------------'
    print 'Hackaday Stats'
    print '----------------------'
    print 'Total profile followers: ' + str(HDstats[0])
    print 'Total likes: ' + str(HDstats[1])
    print 'Total views: ' + str(HDstats[2])
    print 'Total projects followers: ' + str(HDstats[3])
    print 'Total comments: ' + str(HDstats[4])

if YTuser != '':
    YTstats = youtube.getStats(YTuser)
    print ''
    print '----------------------'
    print 'Youtube Stats'
    print '----------------------'
    print 'Total suscriptors: ' + str(YTstats[0])
    print 'Total views: ' + str(YTstats[1])
    print 'Total likes: ' + str(YTstats[2])
    print 'Total dislikes: ' + str(YTstats[3])

if TWuser != '':
    TWstats = twitter.getStats(TWuser)
    print ''
    print '----------------------'
    print 'Twitter Stats'
    print '----------------------'
    print 'Total followers: ' + str(TWstats)
#
if GGname != '':
    GGstats = googlegroups.getStats(GGname)
    print ''
    print '----------------------'
    print 'Thor Google Group Stats'
    print '----------------------'
    print 'Total members: ' + str(GGstats[0])
    print 'Total threads: ' + str(GGstats[1])

if IGuser != '':
    IGstats = instagram.getStats(IGuser)
    print ''
    print '----------------------'
    print 'Instagram Stats'
    print '----------------------'
    print 'Total followers: ' + str(IGstats[0])
    print 'Total likes: ' + str(IGstats[1])
    print 'Total reproductions: ' + str(IGstats[2])
    print 'Total comments: ' + str(IGstats[3])

print ''
print ''
